pkg/apis/core/validation/shoot.go

```diff
@@ -285,7 +285,7 @@ func ValidateShootSpecUpdate(newSpec, oldSpec *core.ShootSpec, newObjectMeta met
 
 	if newObjectMeta.DeletionTimestamp != nil && !apiequality.Semantic.DeepEqual(newSpec, oldSpec) {
 		if diff := deep.Equal(newSpec, oldSpec); diff != nil {
-			return field.ErrorList{field.Forbidden(fldPath, strings.Join(diff, ","))}
+			return field.ErrorList{field.Forbidden(fldPath, fmt.Sprintf("can't edit shoot while it's in deletion. Requested changes: %s", strings.Join(diff, ",")))}
```

Comment:
You could check https://github.com/gardener/gardener/pull/7657.

The same logic exists in many places. Could you please adapt accordingly the rest of the affected occurrences for consistency reasons across the codebase?

cc @acumino